torch/utils/dlpack.py

```diff
@@ -25,3 +15,38 @@
 The dlpack shares the tensors memory.
 Note that each dlpack can only be consumed once.
 """)
+
+
+def from_dlpack(ext_tensor: Any) -> torch.Tensor:
+    """from_dlpack(ext_tensor) -> Tensor
+
+    Convers a tensor from a external library into a ``torch.Tensor``
+    by means of the ``__dlpack__`` protocol.
+
+    The tensor will share the memory with the object represented
```

Comment:
"The tensor and the DLPack capsule share their memory."

Is this true when converting a tensor to a DLPack capsule, too? How does that work if a tensor requires autograd?

Reply:
If converting to a DLPack capsule from a torch tensor is also memory sharing, then I think we should model the operation after `.numpy()`, which does the following when a tensor requires grad:

```
t = torch.tensor((1., 2., 3.), requires_grad=True)
t.numpy()
: RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.
```

See https://github.com/pytorch/pytorch/pull/59790/files for the latest on `.numpy()` semantics. We also don't support tensors with the conjugate bit set, for example.

Reply:
this is a good point